From 12f380f4934dda0da644ee715866865f767dafbc Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Tue, 17 Jan 2017 11:00:13 -0600
Subject: [PATCH 031/205] ARM: OMAP2+: omap_device: Fix HWMOD_NEEDS_REIDLE
 check to actually test

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

Currently there is a check if an omap_hwmod has the HWMOD_NEEDS_REIDLE
flag set in the omap_device core that improperly uses bitwise OR to mask
out the bit rather than bitwise AND, meaning the test will always show
the flag is set whether it is or not. Although this is not causing any
issues today this should be fixed to avoid potential pain in the future.

Fixes: f92a644f7b39 ("ARM: OMAP2+: omap_hwmod: Introduce HWMOD_NEEDS_REIDLE")
Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
(cherry picked from commit c0c6a4cbdd04b780edcf0552dda697a9767d06cc)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/omap_device.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/omap_device.c b/arch/arm/mach-omap2/omap_device.c
index d04e696..f7568ba 100644
--- a/arch/arm/mach-omap2/omap_device.c
+++ b/arch/arm/mach-omap2/omap_device.c
@@ -317,7 +317,7 @@ static int _omap_device_reidle_hwmods(struct omap_device *od)
 	int i;
 
 	for (i = 0; i < od->hwmods_cnt; i++)
-		if (od->hwmods[i]->flags | HWMOD_NEEDS_REIDLE)
+		if (od->hwmods[i]->flags & HWMOD_NEEDS_REIDLE)
 			omap_hwmod_enable_reidle(od->hwmods[i]);
 
 	/* XXX pass along return value here? */
-- 
1.7.9.5

